sdk/include/opentelemetry/sdk/common/http_client.h

```diff
@@ -0,0 +1,129 @@
+#pragma once
+
+#include <chrono>
+#include <string>
+#include <vector>
+#include "opentelemetry/nostd/function_ref.h"
+#include "opentelemetry/version.h"
+
+// Usage Example
+//
+// struct ReponseHandler: public HttpResponseHandler {
+//      void OnHttpResponse(HttpResponse& res) noexcept override
+//      {
+//           if (res.IsSuccess())
+//              .. process response
+//      }
+//  };
+//
+//   HttpSessionManager sessionManager; // implementer can provide singleton implementation for it
+//   auto session = sessionManager.createHttpSession("localhost", 8000);
+//   auto request = session->CreateRequest();
+//   request->AddHeader(..);
+//   ResponseHandler res_handler;
+//   session->SendRequest(res_handler);
+//   session->FinishSession() // optionally in the end.
+//   ...shutdown
+//    sessionManager.FinishAllSessions()
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace common
+{
+namespace http
+{
+
+enum class HttpMethod
+{
+  HTTP_GET,
+  HTTP_POST,
+  HTTP_PUT,
+  HTTP_OPTIONS,
+  HTTP_HEAD,
+  HTTP_PATCH,
+  HTTP_DELETE
+};
+
+enum class HttpSessionState
+{
+  CREATED,  // session object is created
```

Comment:
Follow the Google Naming Convention https://github.com/open-telemetry/opentelemetry-cpp/blob/b395e1afd24bcc22fc07d5e6433f008ce33fa599/CONTRIBUTING.md#development.